src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -821,38 +806,12 @@ private async Task ValidateSignUpPoliciesAsync(Guid ownerId)
         }
     }
 
-    public async Task UpdateLicenseAsync(Guid organizationId, OrganizationLicense license)
+    public async Task UpdateLicenseAsync(Organization organization, OrganizationLicense license)
     {
-        var organization = await GetOrgById(organizationId);
-        if (organization == null)
-        {
-            throw new NotFoundException();
-        }
-
-        if (!_globalSettings.SelfHosted)
-        {
-            throw new InvalidOperationException("Licenses require self hosting.");
-        }
-
-        if (license?.LicenseType != null && license.LicenseType != LicenseType.Organization)
-        {
-            throw new BadRequestException("Premium licenses cannot be applied to an organization. "
-                                          + "Upload this license from your personal account settings page.");
-        }
-
-        if (license == null || !_licensingService.VerifyLicense(license))
-        {
-            throw new BadRequestException("Invalid license.");
-        }
-
-        if (!license.CanUse(_globalSettings))
-        {
-            throw new BadRequestException("Invalid license. Make sure your license allows for on-premise " +
-                "hosting of organizations and that the installation id matches your current installation.");
-        }
+        license.CanUse(_globalSettings, _licensingService);
 
         var enabledOrgs = await _organizationRepository.GetManyByEnabledAsync();
```

Comment:
Past this point, we are validating the license in the context of this particular organization. Not sure whether to leave all these queries bundled with the command or not.